### PR TITLE
fix(ktable): add missing test coverage for data changes as expected [KHCP-7560]

### PR DESCRIPTION
**Pull request by mptap**
Addresses https://konghq.atlassian.net/browse/KHCP-7560

# Summary

Adds missing test for the following scenarios:

- Data changes as expected when clicking a specific page number for non-offset pagination
- Data changes as expected when clicking arrows for offset based pagination
- Data changes as expected when pageSize is changed
- Data changes as expected when sort key or sort direction is changed and NOT using clientSideSort

## PR Checklist

* [x] Does not introduce dependencies
* [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [x] **Tests pass:** check the output of yarn test
* [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [x] **Framework style:** abides by the essential rules in Vue's style guide
* [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: src/components/KTable/KTable.cy.ts [hunk 1]
```diff
@@ -149,6 +149,198 @@ describe('KTable', () => {
...
+      cy.getTestId('k-table-pagination').should('be.visible')
+      cy.getTestId('k-select-input').click()
+      cy.getTestId('k-select-item-1').click({ multiple: true, force: true })
+      cy.getTestId('next-btn').find('a').click()
+      cy.get('.pagination-button.active').should('contain.text', 2 + '')
+      cy.get('.k-table').find('tr').should('have.length', 4)
+    })
+
+    it('when clicking arrows for offset based pagination', () => {
+      const offsetPaginationHeaders = [
```
Comment (kaiarrowood) [inline src/components/KTable/KTable.cy.ts:183]: suggestion: Can you move you mock data logic into a separate file in the mocks folder? https://github.com/Kong/kongponents/tree/main/mocks

File: src/components/KTable/KTable.cy.ts [hunk 2]
```diff
@@ -149,6 +149,198 @@ describe('KTable', () => {
     })
   })
 
+  describe('data changes as expected', () => {
```
Comment (adamdehaven) [inline src/components/KTable/KTable.cy.ts:152]:
```suggestion
  describe('data revalidates and changes as expected', () => {
```

Comment (mptap) [reply src/components/KTable/KTable.cy.ts:152]: updated

File: src/components/KTable/KTable.cy.ts (hunk 1 above)
Comment (mptap) [reply src/components/KTable/KTable.cy.ts:183]: Done

**Review by kaiarrowood — APPROVED** [review]
LGTM! 🚀

**Pull request merged**